lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp

```diff
@@ -6,11 +6,44 @@
 
 namespace mlir::tt::ttir {
 
+bool mock_is_output_tensor_legal_for_op(Operation *op, LayoutAttr layout) {
+  // Placeholder, needs to be replaced with a call the the TTNN op interface.
+  return true;
+}
+
+bool tensor_shape_compatible_with_shard(Operation *op, LayoutAttr layout) {
+  // These constraints are implemented seteratelly in every TTNN op.
+  // Almost nothing seems to be shared between EVERY op, so is hard to have any
+  // logic here without the risk of discarding a valid configuraiton or modeling
+  // the constraint for each op. This logic may be offloaded to the TTNN op
+  // interface.
+
+  // For now we will check if the tilised tensor dims are divisible by the grid
+  // dims. This will definitly discard possible valid configurations, but is a
+  // start.
+  RankedTensorType tensorType =
+      mlir::cast<RankedTensorType>(op->getResult(0).getType());
+  llvm::ArrayRef<int64_t> tensorShape = tensorType.getShape();
+
+  long MTiles = 1;
+  if (tensorType.getRank() >= 2) {
+    MTiles = (tensorShape.rbegin()[1] + 31) / 32;
+  }
+
+  auto KTIles = (tensorShape.back() + 31) / 32;
+
+  auto gridR = layout.getGrid().getShape()[0];
+  auto gridC = layout.getGrid().getShape()[1];
+
+  return (MTiles % gridR == 0) && (KTIles % gridC == 0);
+}
+
 bool LegalGridAnalysis::applyOverrides() {
   // Lookup grid size overrides based on location information for current
   // operation.
   //
 
+  // TODO(odjuricic): We may need to infer shard type.
```

Comment:
Yeah override needs to be extended.

Reply:
Do we want all LayoutAttr params to be overridable?

Reply:
Because of linking with TT-explorer most likely yes, but let's find some good enough starting point.

Reply:
Ok, will add this as separate issue and tackle after this.